pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -171,19 +166,19 @@ def wrap_policy(*args, **kwargs):
             yield
 
         log.detail(f"{self.__class__.__name__}: exiting model_sharded_context.")
+        self.setup_optimizers(self.lightning_module.trainer)
+        optimizers_to_device(self.optimizers, self.root_device)
+        self.setup_precision_plugin()
```

Comment:
why can it not be done in setup? setup() is mean to do exactly that, and setup gets called after the model is wrapped (see trainer.py). I highly recommend to keep it in setup()

Reply:
Yes, I can change that. I needed to double check the ordering around wrapping FSDP vs move to device. 

But looking again at the trainer code, why does the Trainer need to explicitly call configure_sharded_model. Could we move that to the Strategy's setup instead?